backends/arm/test/tester/arm_tester.py

```diff
@@ -61,20 +62,26 @@ def __init__(
                     functional test, no numerical checks.
             profile (TosaProfile): The TOSA profile to use. Either
                 TosaProfile.BI or TosaProfile.MI
+            permute_memory_to_nhwc (bool) : flag for enabling the memory format
+                permutation to nhwc as required by TOSA
         """
         self.tosa_test_util = None
         self.is_quantized = profile == TosaProfile.BI
+        self.permute_memory_to_nhwc = permute_memory_to_nhwc
+
         if backend == ArmBackendSelector.TOSA:
             self.tosa_test_util = TosaTestUtils(profile=profile)
             # The spec below tiggers arm_backend.py to output two files:
             #   1) output.tosa
             #   2) desc.json
             # Saved on disk in self.tosa_test_util.intermediate_path
             self.compile_spec = generate_tosa_compile_spec(
-                self.tosa_test_util.intermediate_path
+                self.permute_memory_to_nhwc, self.tosa_test_util.intermediate_path
```

Comment:
In the long run imagine this to be transparent to the delegate logic. I.e all the delegate does is add `converts` if required. But if you have a graph like,

G1 `input --> nhwc_only_op --> output`
G2 `input --> convert_to_nhwc --> nhwc_only_op --> convert_to_nhwc --> output` added by a pass before `to_backed` or first thing inside `arm_delegate::preprocess()`.
G3 `input --> arm_delegate_op --> output`

Reply:
ack.